lib/internal/mime.js

```diff
@@ -0,0 +1,369 @@
+'use strict';
+
+const {
+  FunctionPrototypeCall,
+  ObjectDefineProperty,
+  RegExpPrototypeExec,
+  SafeMap,
+  SafeStringPrototypeSearch,
+  StringPrototypeCharAt,
+  StringPrototypeIndexOf,
+  StringPrototypeSlice,
+  StringPrototypeToLowerCase,
+  SymbolIterator,
+} = primordials;
+const {
+  ERR_INVALID_MIME_SYNTAX,
+} = require('internal/errors').codes;
+
+const NotHTTPTokenCodePoint = /[^!#$%&'*+\-.^_`|~A-Za-z0-9]/g;
+const NotHTTPQuotedStringCodePoint = /[^\t\u0020-~\u0080-\u00FF]/g;
+
+const END_BEGINNING_WHITESPACE = /[^\r\n\t ]|$/;
+const START_ENDING_WHITESPACE = /[\r\n\t ]*$/;
```

Comment:
Should all constant regexes be in screaming snake case? Or none of them?

```suggestion
const NOT_HTTP_TOKEN_CODE_POINT = /[^!#$%&'*+\-.^_`|~A-Za-z0-9]/g;
const NOT_HTTP_QUOTED_STRING_CODE_POINT = /[^\t\u0020-~\u0080-\u00FF]/g;

const END_BEGINNING_WHITESPACE = /[^\r\n\t ]|$/;
const START_ENDING_WHITESPACE = /[\r\n\t ]*$/;
```